### PR TITLE
[BDTT-11, BDTT-18] BE APIs for Sign Up and Sign In

### PR DESCRIPTION
**Pull request by charmjj**
# Description

Fixes BDTT-11, BDTT-18

- DB: 
1. Modified wguser_id column datatype to varchar2(200),
2. Modified password column to password_hash 
3. Added new password_salt column

- BE APIs for Sign Up and Sign In

- README.md and Requirements.txt files for backend

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] I have added the appropriate labels to my PR
- [x] My code follows the style guidelines of this project
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: backend/app/user.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+from sqlalchemy.ext.compiler import compiles
+from sqlalchemy.exc import IntegrityError
+
+import hashlib
+import re
+import shortuuid
+import uuid
+
+app = Flask(__name__)
+app.config['SQLALCHEMY_DATABASE_URI'] = "oracle://wg:wgdemo*()@202.73.56.175:1521/efprod" # TODO: Externalise into env config file
```
Comment (JeffS97) [inline backend/app/user.py:12]: I dont think should include the username + password in this file, github will flag out as well and spam our emails. Can try adding into env?

**Review by JeffS97 — CHANGES_REQUESTED** [review]

File: backend/app/user.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    email = json_payload['email']
+    password = json_payload['password']
+    phone = json_payload['phone']
+    company = json_payload['company']
+
+    if validate_username(username) is not None:
+        return validate_username(username)
+
+    password_salt = uuid.uuid4().hex
+    password_hash = hashlib.sha512(
```
Comment (JeffS97) [inline backend/app/user.py:85]: Not sure if database can store sha512 hash with varchar(200), need to test and check

File: backend/app/user.py (hunk 1 above)
Comment (charmjj) [reply backend/app/user.py]: done, also updated the README, gitignore and requirements.txt

File: backend/app/user.py (hunk 2 above)
Comment (charmjj) [reply backend/app/user.py:90]: should be ok, 128 digits hexadecimal string < 200. I tested the api and no issues found

**Review by JeffS97 — APPROVED** [review]
Looks good to me

**Pull request merged**